script/Deploy.s.sol

```diff
@@ -4,9 +4,82 @@
 pragma solidity 0.8.20;
 
 import {Script} from "forge-std/Script.sol";
+import {console2} from "forge-std/console2.sol";
+import {CreatorTokenFactory} from "src/CreatorTokenFactory.sol";
+import {IShowtimeVerifier} from "src/lib/IShowtimeVerifier.sol";
 
 contract Deploy is Script {
+  CreatorTokenFactory creatorTokenFactory;
+  uint256 constant EXPECTED_NONCE = 0; // TODO Edit this with the nonce of the deployer address
+    // The list of networks to deploy to.
+  string[] public networks = ["mainnet", "base"];
+  // TODO Edit this with the address of the ShowtimeVerifier contract
+  IShowtimeVerifier[] VERIFIERS = [
+    // ShowtimeVerifier contract on mainnet
+    IShowtimeVerifier(0x481273EB2B6A21e918f6952A6c53C08691FE768F),
+    // ShowtimeVerifier contract on base
+    IShowtimeVerifier(0x481273EB2B6A21e918f6952A6c53C08691FE768F)
+  ];
+  // TODO Edit this with the domain separator of the ShowtimeVerifier contract on mainnet
+  bytes32 constant MAINNET_DOMAIN_SEPARATORS = "MAINNET_DOMAIN_SEPARATORS";
+  // TODO Edit this with the domain separator of the ShowtimeVerifier contract on base
+  bytes32 constant BASE_DOMAIN_SEPARATORS = "BASE_DOMAIN_SEPARATORS";
+  bytes32[] DOMAIN_SEPARATORS;
+
+  mapping(string => address) public creatorTokenFactoryAddresses;
+
+  /// @notice Deploy the contract to the list of networks,
   function run() public {
-    vm.broadcast();
+    // Setup Domain Separators
+    DOMAIN_SEPARATORS[0] = MAINNET_DOMAIN_SEPARATORS;
+    DOMAIN_SEPARATORS[1] = BASE_DOMAIN_SEPARATORS;
+
+    if (DOMAIN_SEPARATORS.length != VERIFIERS.length || DOMAIN_SEPARATORS.length != networks.length)
+    {
+      revert("Check your DOMAIN_SEPARATORS, VERIFIERS, and networks arrays' lengths");
+    }
+
+    // Compute the address the contract will be deployed to
+    address expectedContractAddress = computeCreateAddress(msg.sender, EXPECTED_NONCE);
+    console2.log("Expected contract address: %s", expectedContractAddress);
+
+    // Check if the contract is already deployed on the network, if not, deploy it.
+    for (uint256 i; i < networks.length; i++) {
```

Comment:
Hmm, I don't think we need all this! We're not trying to deploy across N networks simulataneously. The deploy script should just be for one (configurable) network, and the deployer should choose which network (via env var or cl argument) at deploy time.

Reply:
Sounds good, made the changes in the same commit as above: [a137aee](https://github.com/showtime-xyz/creator-tokens/pull/40/commits/a137aee05fb7e11bd3357dec0446190d99c0cd88)